pkg/updater/inflate_test.go

```diff
@@ -0,0 +1,601 @@
+		row      state.Row
+		expected []cell
+	}{
+		{
+			name: "basically works",
+		},
+		{
+			name: "preserve cell ids",
+			row: state.Row{
+				CellIds:  []string{"cell-a", "cell-b"},
+				Icons:    blank(2),
+				Messages: blank(2),
+				Results: []int32{
+					int32(state.Row_PASS), 2,
+				},
+			},
+			expected: []cell{
+				{
+					result: state.Row_PASS,
+					cellID: "cell-a",
+				},
+				{
+					result: state.Row_PASS,
+					cellID: "cell-b",
+				},
+			},
+		},
+		{
+			name: "only finished columns contain icons and messages",
+			row: state.Row{
+				CellIds: blank(8),
+				Icons: []string{
+					"F1", "~1", "~2",
+				},
+				Messages: []string{
+					"fail", "flake-first", "flake-second",
+				},
+				Results: []int32{
+					int32(state.Row_NO_RESULT), 2,
+					int32(state.Row_FAIL), 1,
+					int32(state.Row_NO_RESULT), 2,
+					int32(state.Row_FLAKY), 2,
+					int32(state.Row_NO_RESULT), 1,
+				},
+			},
+			expected: []cell{
+				{},
+				{},
+				{
+					result:  state.Row_FAIL,
+					icon:    "F1",
+					message: "fail",
+				},
+				{},
+				{},
+				{
+					result:  state.Row_FLAKY,
+					icon:    "~1",
+					message: "flake-first",
+				},
+				{
+					result:  state.Row_FLAKY,
+					icon:    "~2",
+					message: "flake-second",
+				},
+				{},
+			},
+		},
+		{
+			name: "find metric name from row when missing",
+			row: state.Row{
+				CellIds:  blank(1),
+				Icons:    blank(1),
+				Messages: blank(1),
+				Results: []int32{
+					int32(state.Row_PASS), 1,
+				},
+				Metric: []string{"found-it"},
+				Metrics: []*state.Metric{
+					{
+						Indices: []int32{0, 1},
+						Values:  []float64{7},
+					},
+				},
+			},
+			expected: []cell{
+				{
+					result: state.Row_PASS,
+					metrics: map[string]float64{
+						"found-it": 7,
+					},
+				},
+			},
+		},
+		{
+			name: "prioritize local metric name",
+			row: state.Row{
+				CellIds:  blank(1),
+				Icons:    blank(1),
+				Messages: blank(1),
+				Results: []int32{
+					int32(state.Row_PASS), 1,
+				},
+				Metric: []string{"ignore-this"},
+				Metrics: []*state.Metric{
+					{
+						Name:    "oh yeah",
+						Indices: []int32{0, 1},
+						Values:  []float64{7},
+					},
+				},
+			},
+			expected: []cell{
+				{
+					result: state.Row_PASS,
+					metrics: map[string]float64{
+						"oh yeah": 7,
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			var actual []cell
+			for r := range inflateRow(context.Background(), &tc.row) {
+				actual = append(actual, r)
+			}
+
+			if !reflect.DeepEqual(actual, tc.expected) {
+				t.Errorf("inflateRow(%v) got %v, want %v", tc.row, actual, tc.expected)
+			}
+		})
+	}
+}
+
+func TestInflateMetic(t *testing.T) {
+	point := func(v float64) *float64 {
+		return &v
+	}
+	cases := []struct {
+		name     string
+		indices  []int32
+		values   []float64
+		expected []*float64
+	}{
+		{
+			name: "basically works",
+		},
+		{
+			name:    "documented example works",
```

Comment:
I think "documented example" should be more descriptive here (and below), it's unclear where the documented examples are and I shouldn't need to cross-reference.